charts/newrelic-infrastructure-v3/templates/control-plane/configmap.yaml

```diff
@@ -8,11 +8,28 @@ metadata:
   name: {{ template "newrelic.fullname" . }}-control-plane
   namespace: {{ .Release.Namespace }}
 data:
-  nri-kubernetes.yml: |
+  nri-kubernetes.yml: |-
     {{- (merge .Values.common.config (include "newrelic.integrationConfigDefaults" . | fromYaml)) | toYaml | nindent 4 }}
     controlPlane:
       enabled: true
-      {{- if .Values.controlPlane.config }}
-      {{- toYaml .Values.controlPlane.config | nindent 6 }}
-      {{- end }}
-{{- end }}
+
+    {{- if  .Values.controlPlane.config.etcd.enabled }}
+      etcd:
+        {{- toYaml .Values.controlPlane.config.etcd | nindent 8 -}}
+    {{- end -}}
+
+    {{- if .Values.controlPlane.config.scheduler.enabled }}
+      scheduler:
+        {{- toYaml .Values.controlPlane.config.scheduler | nindent 8 -}}
+    {{- end -}}
+
+    {{- if  .Values.controlPlane.config.controllerManager.enabled }}
+      controllerManager:
+        {{- toYaml .Values.controlPlane.config.controllerManager | nindent 8 -}}
+    {{- end -}}
+
+    {{- if .Values.controlPlane.config.apiServer.enabled }}
+      apiServer:
+        {{- toYaml .Values.controlPlane.config.apiServer | nindent 8 -}}
+    {{- end -}}
```

Comment:
Is there a reason to expand this by hand?

Reply:
basically to avoid putting in the config useless stuff for disabled component, it is also useful for tests. If needed we can have it back as before